drivers/flash/Kconfig

```diff
@@ -162,4 +162,14 @@ config SOC_FLASH_MCUX_DEV_NAME
 	help
 	  Specify the device name for the flash driver.
 
+config SOC_FLASH_WRITE_BLOCK_SIZE
+	int "write block size (fixed)"
+	depends on FLASH && (SOC_FLASH_NRF5 || SOC_FLASH_STM32 ||\
+	SOC_FLASH_QMSI)
+	range 1 1 if SOC_FLASH_NRF5
+	range 4 4 if SOC_FLASH_STM32
+	range 4 4 if SOC_FLASH_QMSI
```

Comment:
using conditional defaults would be more readable

Reply:
Using keyword 'range' is only way to force the value. So I will keep this as is.